docs: Update Python docs on project management (#8465)
dagger/dagger: docs/current_docs/manuals/developer/ide-integration.mdx

For example, to open the Visual Studio Code from the terminal, with functioning autocompletions (assuming the [Python extensions](https://github.com/microsoft/vscode-python) are installed):

```shell
dagger develop
uv run code .
```

For NeoVim, just replace `code` with `vim`, or any other terminal based editor.

:::note
The `uv run` command works with a `uv.lock` file. If you need to use a `requirements.lock` or other, continue reading for more examples.
:::

## Project environment

The following examples show how to get a working [project environment](https://docs.astral.sh/uv/concepts/projects/#project-environments) (`.venv`) from the terminal, just make sure you run `dagger develop` first, to get the necessary `sdk` directory locally.

<Tabs groupId="pytools">
<TabItem value="uv">

When using a `uv.lock`:

```shell
uv sync
```

If you have an older Dagger module, without a `uv.lock` file, you can create it with:

```shell
uv add --editable ./sdk
rm requirements.lock
````

:::note

> Uv with `uv.lock` needs to have the SDK library (`dagger-io`, generated in `./sdk`) defined as a **production** dependency, unlike the other installation methods where it should be a development dependency.

Reviewer: General comment: if possible, could you please avoid the special formatting like `**` or `*` unless absolutely necessary? Instead, it would be better to use note/tip/warning blocks for highlighting important aspects. The reason is that different authors tend to use this formatting for different purposes, which creates consistency issues, and readers also don't interpret it all the same way, which creates readability issues.

:::

</TabItem>
<TabItem value="uv pip">

> When using a `requirements.lock` file, uv provides a faster ["pip interface"](https://docs.astral.sh/uv/pip/), that replaces multiple tools. It can manage the Python installation, virtual environment creation, and dependency installation.

Reviewer: ```suggestion
When using a `requirements.lock` file, uv provides a faster ["pip interface"](https://docs.astral.sh/uv/pip/) that replaces multiple tools. It can manage the Python installation, virtual environment creation, and dependency installation.
```

```toml
[project]
name = "main"
version = "0.0.0"

[tool.hatch.metadata]
allow-direct-references = true

[tool.hatch.envs.dev]
dependencies = [
  "dagger-io @ {root:uri}/sdk",
]
```

Then, open the editor with the *dev* [Hatch environment](https://hatch.pypa.io/latest/config/environment/overview/):

```shell
hatch run dev:vim .
```

:::note
Hatch manages the project environment in a central location by default instead of `.venv`.
:::

</TabItem>
<TabItem value="poetry">

First, add `./sdk` as a development dependency:

```shell
poetry add --group=dev -e ./sdk
```

Then, open the editor with:

```shell
poetry run vim .
```

:::note
Poetry manages the project environment in a central location instead of `.venv`.
:::

</TabItem>
</Tabs>

:::tip
If you place the virtual environment (`.venv`) inside the module, don't forget to add it to `.gitignore` and to `"exclude"` in `dagger.json` to avoid uploading those files to the runtime container unnecessarily.
:::

## Visual Studio Code

> The simplest way to get [autocompletions and IntelliSense](https://code.visualstudio.com/docs/python/editing#_autocomplete-and-intellisense) working is to open it with the directory hat has `.venv` (next to `.pyproject.toml`), because VS Code picks it up automatically when it's in the root of the [VS Code workspace](https://code.visualstudio.com/docs/python/python-tutorial#_start-vs-code-in-a-workspace-folder):

Reviewer: ```suggestion
The simplest way to get [autocompletions and IntelliSense](https://code.visualstudio.com/docs/python/editing#_autocomplete-and-intellisense) working is to open it with the directory that has `.venv` (next to `.pyproject.toml`), because the IDE picks it up automatically when it's in the root of the [workspace](https://code.visualstudio.com/docs/python/python-tutorial#_start-vs-code-in-a-workspace-folder):
```

```shell
code .            
```

VS Code doesn't need to be opened from the terminal though. The same directory can be opened with VS Code's **File > Open Folder** from the graphical interface (GUI).

The virtual environment can also be created and managed from within the IDE instead of from the terminal, if it supports the chosen lock file format. To know more, see [Visual Studio Code Python Environments](https://code.visualstudio.com/docs/python/environments).

## PyCharm

> PyCharm is a very popular IDE for Python projects. Similarly to VS Code, the simplest way for PyCharm to recognize the Dagger types is to open it on the directory that has `.venv` and `pyproject.toml`, via the terminal or the GUI.

Reviewer: ```suggestion
PyCharm is a very popular IDE for Python projects. The simplest way for PyCharm to recognize the Dagger package is to open it in the directory that has `.venv` and `pyproject.toml`, via the terminal or the GUI.
```

Suggesting package here instead of types, revert if not correct

Author: In that case it should be:

```suggestion
PyCharm is a very popular IDE for Python projects. The simplest way for PyCharm to recognize the `dagger` package is to open the IDE in the directory that has `.venv` and `pyproject.toml`, via the terminal or the GUI.
```
